circleci-artifacts/src/test/java/com/palantir/gradle/utils/circleciartifacts/CircleCiArtifactsTest.java

```diff
@@ -33,39 +34,49 @@ class CircleCiArtifactsTest {
 
     @BeforeEach
     void setup(RootProject rootProject) {
-        rootProject.buildGradle().append("""
-            import com.palantir.gradle.utils.circleciartifacts.CircleCiArtifacts
-
-            public abstract class CircleCiArtifactsTask extends DefaultTask {
-              @Nested
-              abstract CircleCiArtifacts getCircleCiArtifacts();
-            }
-
-            tasks.register("printCircleCiLocation", CircleCiArtifactsTask) {
-                doLast { task ->
-                    def artifactLocation = task.circleCiArtifacts.resolveArtifactLocation('location/in/artifacts')
-                    if (artifactLocation.isPresent()) {
-                        println "Physical path: ${artifactLocation.get().physicalPath()}"
-                        println "External location: ${artifactLocation.get().externalLocation()}"
-                        println "Circle link: ${artifactLocation.get().circleLink()}"
-                    } else {
-                        println "Not in Circle, empty artifact location"
-                    }
-                }
-            }
-            """);
+        rootProject
+                .buildGradle()
+                .prepend(
+                        """
+                        buildscript {
+                            repositories {
+                                mavenLocal()
+                            }
+                            dependencies {
+                                classpath 'com.palantir.gradle.utils:circleci-artifacts:%s'
+                            }
+                        }
+
+                        import com.palantir.gradle.utils.circleciartifacts.CircleCiArtifacts
+
+                        public abstract class CircleCiArtifactsTask extends DefaultTask {
+                          @Nested
+                          abstract CircleCiArtifacts getCircleCiArtifacts();
+                        }
+
+                        tasks.register("printCircleCiLocation", CircleCiArtifactsTask) {
+                            doLast { task ->
+                                def artifactLocation = task.circleCiArtifacts.resolveArtifactLocation('location/in/artifacts')
+                                if (artifactLocation.isPresent()) {
+                                    println "Physical path: ${artifactLocation.get().physicalPath()}"
+                                    println "External location: ${artifactLocation.get().externalLocation()}"
+                                    println "Circle link: ${artifactLocation.get().circleLink()}"
+                                } else {
+                                    println "Not in Circle, empty artifact location"
+                                }
+                            }
+                        }
+                        """,
+                        Optional.ofNullable(System.getProperty("projectVersion"))
```

Comment:
nit - this is sort of hidden when just put as a parameter to a string format.  I'd make it a local variable just so it stands out.